src/remote_control.cpp

```diff
@@ -50,18 +50,28 @@ namespace rviz_visual_tools
 /**
  * \brief Constructor
  */
-RemoteControl::RemoteControl(const ros::NodeHandle& nh) : nh_(nh)
+RemoteControl::RemoteControl(const ros::NodeHandle& nh)
+: nh_{ nh },
+  spinner_{ 1, &callback_queue_ }
 {
   std::string rviz_dashboard_topic = "/rviz_visual_tools_gui";
 
   // Subscribe to Rviz Dashboard
   const std::size_t button_queue_size = 10;
+
+  nh_.setCallbackQueue(&callback_queue_);
+
   rviz_dashboard_sub_ = nh_.subscribe<sensor_msgs::Joy>(rviz_dashboard_topic, button_queue_size,
                                                         &RemoteControl::rvizDashboardCallback, this);
-
+  spinner_.start();
   ROS_INFO_STREAM_NAMED(name_, "RemoteControl Ready.");
 }
 
+RemoteControl::~RemoteControl() {
+  // stop spinner explicitly for clarity.
```

Comment:
strictly speaking this is not necessary because `spinner_` is the last of the newly added members in the class and thus the first to be destroyed. I still prefer it here.

It would be nice to have a class `SpinningSubscriber` somewhere for this use-case..